pkg/skaffold/deploy/deploy.go

```diff
@@ -39,3 +39,28 @@ type Deployer interface {
 	// Cleanup deletes what was deployed by calling Deploy.
 	Cleanup(context.Context, io.Writer) error
 }
+
+type Result struct {
+	err        error
+	namespaces []string
+}
+
+func NewDeployErrorResult(err error) *Result {
+	return &Result{err: err}
+}
+
+func NewDeploySuccessResult(namespaces []string) *Result {
+	return &Result{namespaces: namespaces}
+}
+
+func (d *Result) Namespaces() []string {
+	return d.namespaces
+}
+
+func (d *Result) IsError() (bool, error) {
```

Comment:
nit: I feel like I typically see "OK bools" returned as the second argument-- 

for example, with maps:
```go
val, ok := myMap[key]
```